ci: Re-build when docker images aren't available

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #78186

Adds a failsafe for when docker images aren't available at your base at
the expense of TTS. Also adds the ability to set force_push and note
about what to do when TTS gets out of hand because docker images are
always re-building.

Fixes https://github.com/pytorch/pytorch/issues/70881

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
ci: Re-build when docker images aren't available

Adds a failsafe for when docker images aren't available at your base at
the expense of TTS. Also adds the ability to set force_push and note
about what to do when TTS gets out of hand because docker images are
always re-building.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented on .circleci/docker/build_docker.sh:
```diff
@@ -46,7 +46,9 @@ fi
 ./build.sh ${IMAGE_NAME} -t "${image}:${tag}"
 
-docker push "${image}:${tag}"
+if [ "${DOCKER_SKIP_PUSH:-false}" = "true" ]; then
```
This should probably be reversed to `DOCKER_ALLOW_PUSH` but I wanted to keep the default behavior of the script just in case there's others that rely on the behavior of this script

seemethere committed:
Update on "ci: Re-build when docker images aren't available"


Adds a failsafe for when docker images aren't available at your base at
the expense of TTS. Also adds the ability to set force_push and note
about what to do when TTS gets out of hand because docker images are
always re-building.

Fixes https://github.com/pytorch/pytorch/issues/70881

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Re-build when docker images aren't available"


Adds a failsafe for when docker images aren't available at your base at
the expense of TTS. Also adds the ability to set force_push and note
about what to do when TTS gets out of hand because docker images are
always re-building.

Fixes https://github.com/pytorch/pytorch/issues/70881

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented on .github/actions/calculate-docker-image/action.yml:
```diff
@@ -86,6 +96,7 @@ runs:
       env:
         IMAGE_NAME: ${{inputs.docker-image-name}}
         DOCKER_SKIP_S3_UPLOAD: "1"
+        DOCKER_SKIP_PUSH: ${{ steps.check.outputs.skip_push || 'false' }}
```
Setting a default to `false` here since we'd like to keep the legacy behavior here and only want to skip push in the case of the stampeding herd example

janeyx99 commented on .github/actions/calculate-docker-image/action.yml:
```diff
@@ -53,6 +56,7 @@ runs:
         BASE_REVISION: ${{ github.event.pull_request.base.sha || github.sha }}
         DOCKER_IMAGE: ${{ steps.calculate-tag.outputs.docker-image }}
         DOCKER_TAG: ${{ steps.calculate-tag.outputs.docker-tag }}
+        DOCKER_FORCE_PUSH: ${{ inputs.skip_push }}
```
oh lol--i see you meant to put force_push

janeyx99 reviewed (approved):
Looks great! But please address the force push skip push mixup I commented about before merging

seemethere committed:
Update on "ci: Re-build when docker images aren't available"


Adds a failsafe for when docker images aren't available at your base at
the expense of TTS. Also adds the ability to set force_push and note
about what to do when TTS gets out of hand because docker images are
always re-building.

Fixes https://github.com/pytorch/pytorch/issues/70881

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented:
@pytorchbot merge this -g

seemethere commented:
@pytorchbot merge -g
